_source/_posts/2024-04-30-express-universal-logout.md

```diff
@@ -0,0 +1,674 @@
+---
+layout: blog_post
+title: "How to Instantly Sign a User Out across All Your Apps"
+author: semona-igama
+by: advocate
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: "Sign users out across all apps with a universal logout endpoint."
+tags: [enterprise-ready-workshops, express, oidc, universal-logout]
+tweets:
+- ""
+- ""
+- ""
+image: blog/express-universal-logout/social.jpg
+type: awareness
+github: https://github.com/oktadev/okta-enterprise-ready-workshops/tree/ul-workshop-complete
+---
+
+Your enterprise customers expect you to safeguard them from common security incidents, especially when it comes to compromised user accounts. Perhaps a user has signed in from a known stolen device or another country outside the list of allowed IP zones. If a hacker is masquerading as one of your customer's employees, potentially accessing sensitive company data, you must end their session and log them out of your app immediately. 
+
+Bottom line, if you build SaaS applications for enterprise-level customers who leverage Identity Providers (IdPs), workflows, and threat-detection tools, then adding Universal Logout to your app is the solution to ending suspicious user sessions ASAP. 
+
+In this tutorial, you will learn to add a secure Universal Logout API endpoint to a web app, test it by sending a request to end a user's active session, and finally handle signing them out of the app. Lastly, we'll use Okta as the IdP to initiate user logout.
+
+>**Note**: You can follow along with this blog to build a Universal Logout or check out the finished code on the [ul-workshop-complete](https://github.com/oktadev/okta-enterprise-ready-workshops/tree/ul-workshop-complete) branch of our Oktadev GitHub repository.
+
+{% include toc.md %}
+
+## Get the sample app up and running
+
+**Prerequisites**
+- Enterprise Ready OIDC Workshop - [oidc-workshop-complete](https://github.com/oktadev/okta-enterprise-ready-workshops/tree/oidc-workshop-complete)
+- Code Editor (I used [Visual Studio Code](https://code.visualstudio.com/download)) 
+- [Okta Developer Account](https://developer.okta.com/signup/)
+- [Node.js](https://nodejs.org/en) v18 or greater
+
+Follow [these setup instructions](/blog/2023/07/27/enterprise-ready-getting-started) to install and run the Todo sample app. Run `node -v` and ensure you have Node version 18 or newer. 
```

Comment:
There is a strict engine check in the package.json to ensure Node version. If you want to have this disclaimer, it should be prior to the first sentence.